crates/ruff_python_ast/src/lib.rs

```diff
@@ -80,13 +81,25 @@ pub enum PySourceType {
     Ipynb,
 }
 
+impl PySourceType {
+    /// Infers the source type from the file extension.
+    ///
+    /// Falls back to `Python` if the extension is not recognized.
```

Comment:
As mentioned above, this does not seem to me like desirable behavior. Even if we do want this behavior at the user level, it still seems like surprising and undesirable behavior at the level of a library method like this to return a default that is a total guess and hides the actual nature of the underlying data. Better to return `None` or `PySourceType::Unknown` and let the caller decide what default they want.

Reply:
I don't plan on changing this, as this matches Ruff's current behavior.